jwst/ami/mask_definition_ami.py

```diff
@@ -0,0 +1,127 @@
+#
+#  Module for defining mask geometry in pupil space
+#  Replace old mask_definitions by reading info in nrm_model into a similar object.
+#
+
+import numpy as np
+import math
+
+from .utils import rotate2dccw
+
+m = 1.0
+mm = 1.0e-3 * m
+um = 1.0e-6 * m
+
+
+class NRMDefinition():
+
+	def __init__(self, nrm_model, maskname='jwst_ami', rotdeg=None, chooseholes=None):
+		"""
+		Set attributes of NRMDefinition class. 
+
+		Get hole centers and other mask geometry details from NRMModel, apply rotations/flips 
+		as necessary and set them as attributes.
+
+		Parameters
+		----------
+		nrm_model: NRMModel
+			datamodel containing NRM reference file data
+		maskname: string
+			Identifier for mask geometry; default 'jwst_ami', optional
+		rotdeg: list of floats
+			range of rotations to search (degrees), optional
+		chooseholes: list
+			None, or e.g. ['B2', 'B4', 'B5', 'B6'] for a four-hole mask, optional
+			If None, use real seven-hole mask
+		"""
+
+		if maskname not in ['jwst_ami','jwst_g7s6c']:
+			raise ValueError("Mask name not supported")
+
+		self.maskname = maskname # there's only one mask but this is used in oifits
+		self.hdia = nrm_model.flat_to_flat
+		self.activeD = nrm_model.diameter
+		self.OD = nrm_model.pupil_circumscribed
+		self.ctrs = []
+
+		self.read_nrm_model(nrm_model, chooseholes=chooseholes)
+
+		if rotdeg is not None:
+			self.rotdeg = rotdeg
```

Comment:
is there anything wrong with setting `self.rotdeg = None`?  Personally I feel it's easier to check e.g. `if nrmdef.rotdeg is None` rather than `if not hasattr(nrmdef, "rotdeg")`. 

Actually, just searching through these changes (and jwst main) with cmd+F, I don't see this attribute accessed anywhere.  Is it ever used?

Reply:
You're right, looks like that isn't used anymore either.